src/schemes/local.ts

```diff
@@ -36,39 +40,61 @@ const DEFAULTS: SchemeOptions = {
 }
 
 export default class LocalScheme extends BaseScheme<typeof DEFAULTS> {
-  requestHandler: RequestHandler
+  public token: Token
+  public requestHandler: RequestHandler
 
   constructor ($auth, options, ...defaults) {
     super($auth, options, ...defaults, DEFAULTS)
 
+    // Initialize Token instance
+    this.token = new Token(this, this.$auth.$storage)
+
     // Initialize Request Interceptor
-    this.requestHandler = new RequestHandler(this.$auth)
+    this.requestHandler = new RequestHandler(this, this.$auth.ctx.$axios)
   }
 
   _updateTokens (response) {
     if (this.options.token.required) {
       const token = getResponseProp(response, this.options.token.property)
-      this.$auth.token.set(token)
+      this.token.set(token)
     }
   }
 
-  _checkStatus () {
-    if (this.options.token.required) {
-      // Sync token
-      this.$auth.token.sync()
+  check (checkStatus = false, tokenCallback?, _refreshTokenCallback?) {
```

Comment:
This callback-based approach is adding again a dependency complexity and making function, not pure check. What if we instead, always return an object like `{ valid: bool, tokenExpired: true }` this way consumers can take extra actions based on validation feedback.

Reply:
Improved in commit 4d014e13